Code/src/main/java/nl/utwente/viskell/ui/components/DrawWire.java

```diff
@@ -0,0 +1,124 @@
+    
+    /**
+     * @param pane The Pane this wire is on.
+     * @param anchor The starting anchor of new wire.
+     */
+    public DrawWire(CustomUIPane pane, ConnectionAnchor anchor) {
+        this.loadFXML("DrawWire");
+        TactilePane.setDraggable(this, false);
+        TactilePane.setGoToForegroundOnContact(this, false);
+        this.setMouseTransparent(true);
+
+        this.pane = pane;
+        this.anchor = anchor;
+        pane.getChildren().add(this);
+        Point2D initPos = pane.sceneToLocal(anchor.localToScene(anchor.getLocalCenter()));
+        this.setFreePosition(initPos);
+        this.invalidateAnchorPosition();
+        anchor.localToSceneTransformProperty().addListener(this);
+    }
+
+    /**
+     * Constructs a new Connection from this partial wire and another anchor
+     * @param target the Anchor to which the other end of this should be connection to.
+     * @return the newly build Connection or null if it's not possible
+     */
+    public Connection buildConnectionTo(ConnectionAnchor target) {
+        InputAnchor sink;
+        OutputAnchor source;
+        if (this.anchor instanceof InputAnchor) {
+            if (target instanceof InputAnchor) {
+                return null;
+            }
+            sink = (InputAnchor)this.anchor;
+            source = (OutputAnchor)target;
+        } else {
+            if (target instanceof OutputAnchor) {
+                return null;
+            }
+            sink = (InputAnchor)target;
+            source = (OutputAnchor)this.anchor;
+        }
+        
+        if (!target.canAddExtraConnection()) {
+            target.removeConnections(); // push out the existing connections
+        }
+
+        return new Connection(this.pane, source, sink);
+    }
+
+    /** Removes this wire from its pane, and its listener. */
+    public final void remove() {
+        this.anchor.localToSceneTransformProperty().removeListener(this);
+        this.pane.getChildren().remove(this);
+    }
+
+    @Override
+    public void changed(ObservableValue<? extends Transform> observable, Transform oldValue, Transform newValue) {
+        this.invalidateAnchorPosition();
+    }
+
+    /** Update the UI position of the anchor. */
+    private void invalidateAnchorPosition() {
+        Point2D point = pane.sceneToLocal(this.anchor.localToScene(this.anchor.getLocalCenter()));
+        this.setStartX(point.getX());
+        this.setStartY(point.getY());
+        this.updateBezierControlPoints();
+    }
+    
+    /**
+     * Sets the free end coordinates for this wire.
+     * @param point coordinates local to this wire's parent.
+     */
+    public void setFreePosition(Point2D point) {
+        this.setEndX(point.getX());
+        this.setEndY(point.getY());
+        this.updateBezierControlPoints();
+    }
+
+    /** Updates the Bezier offset (curviness) according to the current start and end positions. */
+    private void updateBezierControlPoints() {
```

Comment:
It is intentionally different, but i'm not satisfied with the visual aspects yet.